src/components/reviewMatches/ReviewLink.tsx

```diff
@@ -0,0 +1,318 @@
+import { Container, Divider, Stack, Typography } from '@mui/material'
+import { MakeGenerics, useNavigate, useSearch } from '@tanstack/react-location'
+import { useMutation, useQuery } from '@tanstack/react-query'
+import { AxiosError } from 'axios'
+import { useSnackbar } from 'notistack'
+import { useState } from 'react'
+import { useAppConfig } from '../../hooks/useAppConfig'
+import ApiClient from '../../services/ApiClient'
+import { NotificationState } from '../../types/Notification'
+import { AnyRecord } from '../../types/PatientRecord'
+import Loading from '../common/Loading'
+import ApiErrorMessage from '../error/ApiErrorMessage'
+import NotFound from '../error/NotFound'
+import Button from '../shared/Button'
+import PageHeader from '../shell/PageHeader'
+import DataGrid from './DataGrid'
+import Dialog from './Dialog'
+
+type ReviewLinkParams = MakeGenerics<{
+  Search: {
+    payload: {
+      notificationId: string
+      patient_id: string
+      golden_id: string
+      score: number
+      candidates: { golden_id: string; score: number }[]
+    }
+  }
+}>
+
+//TODO Move horrible function to the backend
+const mapDataToScores = (
+  data?: AnyRecord[],
+  candidates?: { golden_id: string; score: number }[]
+): AnyRecord[] => {
+  if (!data?.length) {
+    return []
+  }
+
+  for (let i = 0; i < data.length; i++) {
+    data[i].score =
+      candidates?.find(c => c.golden_id === data[i].uid)?.score || 0
+  }
```

Comment:
Please use map instead.

Reply:
Is this function used elsewhere ?

Reply:
Doesn't map return an array? And we are looking for a specific value here?